micrometer-observation/src/main/java/io/micrometer/observation/aop/ObservedAspect.java

```diff
@@ -141,6 +149,11 @@ public ObservedAspect(ObservationRegistry registry,
     private @NullUnmarked Object observe(ProceedingJoinPoint pjp, Method method, Observed observed) throws Throwable {
         Observation observation = ObservedAspectObservationDocumentation.of(pjp, observed, this.registry,
                 this.observationConvention);
+
+        if (observedKeyValueAnnotationHandler != null) {
+            observedKeyValueAnnotationHandler.addAnnotatedParameters(observation.getContext(), pjp);
```

Comment:
Here's where we could add support for method result: `addAnnotatedMethodResult`.